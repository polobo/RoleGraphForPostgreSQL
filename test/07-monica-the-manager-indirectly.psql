-- Monica the Service Manager, can get Adminstration on Advisors
-- but otherwise simply has standard grants to advisors and technicians.

\if :is_su
    SELECT format('CREATE USER u7_monica_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('CREATE GROUP g7a_service_manager_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('CREATE GROUP g7b_service_advisor_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('CREATE GROUP g7c_service_tech_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('GRANT g7a_service_manager_%1$s TO u7_monica_%1$s;', :'ctx') \gexec
    SELECT format('GRANT g7b_service_advisor_%1$s TO g7a_service_manager_%1$s WITH ADMIN OPTION;', :'ctx') \gexec
    SELECT format('GRANT g7c_service_tech_%1$s TO g7b_service_advisor_%1$s;', :'ctx') \gexec
\endif
