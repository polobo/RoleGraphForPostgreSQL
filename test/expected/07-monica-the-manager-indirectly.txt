                                                                                                                             Test Result for 07-monica-the-manager-indirectly
        rolname         |                         administration                         | memberof_users |                         memberof_groups                          |                       member_users                        |                          member_groups                          
------------------------+----------------------------------------------------------------+----------------+------------------------------------------------------------------+-----------------------------------------------------------+-----------------------------------------------------------------
 g7a_service_manager_su | of g7b_service_advisor_su from superuser                       |                | g7b_service_advisor_su from superuser (s)                       +| u7_monica_su from superuser (s)                           | 
                        |                                                                |                | g7c_service_tech_su via g7b_service_advisor_su/superuser (s)     |                                                           | 
 g7b_service_advisor_su | by u7_monica_su via g7a_service_manager_su/superuser          +|                | g7c_service_tech_su from superuser (s)                           | u7_monica_su via g7a_service_manager_su/superuser (s)     | g7a_service_manager_su from superuser (s)
                        | by g7a_service_manager_su from superuser                       |                |                                                                  |                                                           | 
 g7c_service_tech_su    |                                                                |                |                                                                  | u7_monica_su via g7b_service_advisor_su/superuser (s)[+1] | g7a_service_manager_su via g7b_service_advisor_su/superuser (s)+
                        |                                                                |                |                                                                  |                                                           | g7b_service_advisor_su from superuser (s)
 cr_admin               | Create Role                                                    |                |                                                                  |                                                           | 
 u7_monica_su           | of g7b_service_advisor_su via g7a_service_manager_su/superuser |                | g7a_service_manager_su from superuser (s)                       +|                                                           | 
                        |                                                                |                | g7b_service_advisor_su via g7a_service_manager_su/superuser (s) +|                                                           | 
                        |                                                                |                | g7c_service_tech_su via g7b_service_advisor_su/superuser (s)[+1] |                                                           | 
(5 rows)

